modules/ldap_query/lambda/lambda.py

```diff
@@ -0,0 +1,467 @@
+    template = env.get_template(template)
+    output_from_parsed_template = template.render(**kwargs)
+    return output_from_parsed_template
+
+
+def create_html_table(content):
+    # un-group the users for the html table
+    template_contents = {
+        "table_headers": get_html_table_headers(content),
+        "user_list": content
+    }
+    artifact = {}
+    artifact['content'] = render_template(**template_contents)
+    artifact['file_name'] = get_file_name("user_expiration", "html")
+    return artifact
+
+
+def generate_artifacts(content):
+    """Returns the list of objects to upload to s3"""
+    artifacts = {}
+    artifacts['user_expiration_table'] = create_json_doc(content)
+    artifacts['user_expiration_html'] = create_html_table(content)
+    return artifacts
+
+
+def put_object(dest_bucket_name, dest_object_name, src_data):
+    """
+    Add an object to an Amazon S3 bucket
+    """
+
+    # Construct Body= parameter
+    if isinstance(src_data, bytes):
+        object_data = src_data
+    else:
+        log.error(
+            f"Type of {str(type(src_data))}"
+            f" for the argument \'src_data\' is not supported.")
+        return False
+
+    # Put the object
+    s3 = boto3.client('s3')
+    # log.debug(f"destination object name: {dest_object_name}")
+    try:
+        s3.put_object(
+            Bucket=dest_bucket_name,
+            ACL="private",
+            ContentEncoding="utf-8",
+            Key=dest_object_name,
+            Body=object_data
+            )
+    except s3.exceptions.ClientError as e:
+        # AllAccessDisabled error == bucket not found
+        # NoSuchKey or InvalidRequest
+        # error == (dest bucket/obj == src bucket/obj)
+        log.error(e)
+        return False
+    finally:
+        if isinstance(src_data, str):
+            object_data.close()
+    return True
+
+
+def create_presigned_url(bucket_name, object_name, expiration=3600):
+    s3 = boto3.client('s3')
+    try:
+        response = s3.generate_presigned_url(
+            'get_object',
+            Params={
+                'Bucket': bucket_name,
+                'Key': object_name
+                },
+            ExpiresIn=expiration
+        )
+    except s3.exceptions.ClientError as e:
+        log.error(e)
+        return None
+    # The response contains the presigned URL
+    return response
+
+
+def upload_artifact(artifact):
+    """Uploads an artifact to s3 and generates a presigned url
+
+    Arguments:
+        artifact {dict} -- dictionary containing the artifact's
+        contents and file name
+
+    Returns:
+        dict -- dictionary containing the object name and presigned url
+    """
+    bucket_name = os.environ['ARTIFACTS_BUCKET']
+    log.debug(f'Uploading object: {artifact["file_name"]} to {bucket_name}')
+    if put_object(
+            bucket_name,
+            artifact["file_name"],
+            artifact["content"].encode("utf-8")):
+        presigned_url = create_presigned_url(
```

Comment:
What if this returns none, because the try/except didn't allow the exception to raise?

Reply:
removed the try/except in the s3 upload function so this should be good now